src/SW_Domain.c

```diff
@@ -0,0 +1,175 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "include/SW_Domain.h"
+#include "include/filefuncs.h"
+#include "include/SW_Files.h"
+#include "include/SW_Times.h"
+
+#ifdef SWNETCDF
+#include "include/SW_netCDF.h"
+#endif
+
+/* =================================================== */
+/*             Local Function Definitions              */
+/* --------------------------------------------------- */
+
+/**
+ * @brief Read `domain.in` and report any problems encountered when doing so
+ *
+ * @param[in] PathInfo Struct holding all information about the programs path/files
+ * @param[out] SW_Domain Struct of type SW_DOMAIN holding constant
+ *      temporal/spatial information for a set of simulation runs
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void SW_DOM_read(PATH_INFO* PathInfo, SW_DOMAIN* SW_Domain, LOG_INFO* LogInfo) {
+
+    FILE *f;
+    int y, keyID;
+    char inbuf[MAX_FILENAMESIZE], *MyFileName;
+    TimeInt tempdoy;
+    char key[10], value[6]; // 10 - Max key size, 6 - max characters
+    Bool fstartdy = swFALSE, fenddy = swFALSE;
+
+    MyFileName = PathInfo->InFiles_csv[eDomain];
+	f = OpenFile(MyFileName, "r", LogInfo);
+
+    // Set SW_DOMAIN
+    while(GetALine(f, inbuf)) {
+        sscanf(inbuf, "%s %s", key, value);
+
+        keyID = domain_inkey_to_id(key);
+        switch(keyID) {
+            // Only expect domain information when SWNETCDF is defined
+            #ifdef SWNETCDF
```

Comment:
I find it confusing that `SW_DOM_read()` is not reading in domain information `#ifdef SWNETCDF` (lines 45-63) -- we just established that "domain.in" is a new text-based input for SOILWAT2 in #365.

I think we should remove those - then the else-branch on lines 126-130 becomes superfluous too (see comment https://github.com/DrylandEcology/SOILWAT2/pull/366/files#r1290876867).